account_setup.py

```diff
@@ -0,0 +1,348 @@
+        parameters=StorageAccountCreateParameters(
+            sku=Sku(SkuName.standard_lrs),
+            kind=Kind.storage,
+            location=kwargs.get('region', DefaultSettings.region)
+        )
+    )
+    return storage_account.result().id
+
+
+
+def create_batch_account(credentials, subscription_id, **kwargs):
+    """
+        Create a Batch account
+        :param credentials: msrestazure.azure_active_directory.AdalAuthentication
+        :param subscription_id: str
+        :param **resource_group: str
+        :param **batch_account: str
+        :param **region: str
+        :param **storage_account_id: str
+    """
+    batch_management_client = BatchManagementClient(credentials, subscription_id)
+    batch_account = batch_management_client.batch_account.create(
+        resource_group_name=kwargs.get("resource_group", DefaultSettings.resource_group),
+        account_name=kwargs.get("batch_account", DefaultSettings.batch_account),
+        parameters=BatchAccountCreateParameters(
+            location=kwargs.get('region', DefaultSettings.region),
+            auto_storage=AutoStorageBaseProperties(
+                storage_account_id=kwargs.get('storage_account_id', DefaultSettings.region)
+            )
+        )
+    )
+    return batch_account.result().id
+
+
+def create_vnet(credentials, subscription_id, **kwargs):
+    """
+        Create a Batch account
+        :param credentials: msrestazure.azure_active_directory.AdalAuthentication
+        :param subscription_id: str
+        :param **resource_group: str
+        :param **virtual_network_name: str
+        :param **subnet_name: str
+        :param **region: str
+    """
+    network_client = NetworkManagementClient(credentials, subscription_id)
+    resource_group_name = kwargs.get("resource_group", DefaultSettings.resource_group)
+    virtual_network_name = kwargs.get("virtual_network_name", DefaultSettings.virtual_network_name)
+    subnet_name = kwargs.get("subnet_name", DefaultSettings.subnet_name)
+    # get vnet, and subnet if they exist
+    virtual_network = subnet = None
+    try:
+        virtual_network = network_client.virtual_networks.get(
+            resource_group_name=resource_group_name,
+            virtual_network_name=virtual_network_name,
+        )
+    except CloudError as e:
+        pass
+
+    if virtual_network:
+        confirmation_prompt = "A virtual network with the same name ({}) was found. \n"\
+                             "Please note that the existing address space and subnets may be changed or destroyed. \n"\
+                             "Do you want to use this virtual network? (y/n): ".format(virtual_network_name)
+        deny_error = AccountSetupError("Virtual network already exists, not recreating.")
+        unrecognized_input_error = AccountSetupError("Input not recognized.")
+        prompt_for_confirmation(confirmation_prompt, deny_error, unrecognized_input_error)
```

Comment:
Never used this before, but does this exit the method?

Reply:
it will raise an error if either the user declines, or if in 3 tries the input is not recognized. Otherwise this returns none and just continues.